pkg/reconciler/managed/reconciler.go

```diff
@@ -789,6 +795,8 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	log.Debug(observation.ObservationMessage)
```

Comment:
```suggestion
	if observation.Diff != "" }
		log.Debug("External resource differs from desired state", "diff", observation.Diff)
	}
```

I'd prefer something like this - i.e. don't log unless the Diff is supplied, and use structured logging.

Reply:
Sounds good. This makes sense as well. Also prevents a bunch of empty line messages for resources that don't have the cmp.Diff logged.